lib/jekyll/page.rb

```diff
@@ -119,7 +119,7 @@ def url_placeholders
     # Returns nothing.
     def process(name)
       self.ext = File.extname(name)
-      self.basename = name[0..-ext.length - 1]
+      self.basename = name[0..-ext.length - 1].gsub(%r!\.+\z!, "")
```

Comment:
You might be able to use `.chomp(".")` here instead. [String#chomp](https://ruby-doc.org/core-2.2.0/String.html#method-i-chomp)

Reply:
I intentionally did not use `String#chomp` here because `#chomp` removes only one trailing character.
```ruby
"hello-world...".chomp(".")           # => "hello-world.."
"hello-world...".gsub(%r!\.*\z!, "")  # => "hello-world"
```